lightning/src/routing/router.rs

```diff
@@ -181,7 +296,27 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, targ
 		return Err(LightningError{err: "Cannot generate a route of more value than all existing satoshis".to_owned(), action: ErrorAction::IgnoreError});
 	}
 
-	// We do a dest-to-source Dijkstra's sorting by each node's distance from the destination
+	// The general routing idea is the following:
+	// 1. Fill first/last hops communicated by the caller.
+	// 2. Attempt to construct a path from source to dest for transferring any ~sufficient (described later) value.
+	//    If succeed, remember which channels were used and how much liquidity they have available,
+	//    so that future paths don't rely on the same liquidity.
+	// 3. Prooceed to the next step if:
+	//    - we hit the recommended target value;
+	//    - OR if we could not construct a new path. Any next attempt to construct it will fail too.
+	//    Otherwise, repeat step 2.
+	// 4. See if we managed to collect paths which aggregately are able to transfer target value
+	//    (not recommended value). If yes, proceed. If not, fail routing.
+	// 5. At this point, the found paths may be transferring more value than we need to send,
+	//    since recommended value is higher than the the value we want to transfer.
+	// 6. Of all the found paths, select only those with the lowest total fee.
+	// 7. Randomly combine paths into routes having enough to fulfill the payment. (TODO: knapsack?)
+	// 8. The last path in every selected route is likely to be more than we need (see step 5).
+	//    Reduce its value-to-transfer and recompute fees.
+	// 9. Choose the best route by the lowest total fee.
```

Comment:
What do you think as a practice to numerote step at each code segment as Val as done for this test : https://github.com/rust-bitcoin/rust-lightning/pull/611/files#diff-b30410f22a759d5e664e05938af7ef2edd244c8a7872e7ada376055ff130088b

Reply:
Good suggestion, I also considered it. Perhaps I will update my code with this idea.